MobiusCore/Source/EventSources/CompositeEventSourceBuilder.swift

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2019 Spotify AB.
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/// A `CompositeEventSourceBuilder` gathers the provided event sources together and builds a single event source that
+/// subscribes to all of them when its `subscribe` method is called.
+public struct CompositeEventSourceBuilder<Event> {
+    private let eventSources: [AnyEventSource<Event>]
+
+    /// Initializes a `CompositeEventSourceBuilder`.
+    public init() {
+        self.init(eventSources: [])
+    }
+
+    private init(eventSources: [AnyEventSource<Event>]) {
+        self.eventSources = eventSources
+    }
+
+    /// Returns a new `CompositeEventSourceBuilder` with the specified event source added to it.
+    public func addEventSource<Source: EventSource>(_ source: Source)
+    -> CompositeEventSourceBuilder<Event> where Source.Event == Event {
+        let es = AnyEventSource<Event>(source)
+        let sources = eventSources + [es]
+        return CompositeEventSourceBuilder(eventSources: sources)
+    }
+
+    /// Builds an event source that composes all the event sources that have been added to the builder.
+    ///
+    /// - Returns: An event source which represents the composition of the builder’s input event sources. The type
+    /// of this source is an implementation detail; consumers should avoid spelling it out if possible.
```

Comment:
In particular, I’d like the return type to be `some EventSource<.Event == Event>`, but Swift 5.1 will only have unconstrained opaque types.